src/BloomExe/web/EnhancedImageServer.cs

```diff
@@ -216,31 +338,38 @@ private string GetUiLanguageFileVersion(string englishFileName, string langCode)
 
 		private bool CheckForSampleTextChanges(IRequestInfo info)
 		{
-			if (_sampleTextsWatcher == null)
+			lock (SyncObj)
 			{
-				var path = Path.Combine(Path.GetDirectoryName(CurrentCollectionSettings.SettingsFilePath), "Sample Texts");
-				if (!Directory.Exists(path))
-					Directory.CreateDirectory(path);
-
-				_sampleTextsWatcher = new FileSystemWatcher {Path = path};
-				_sampleTextsWatcher.Created += SampleTextsOnChange;
-				_sampleTextsWatcher.Changed += SampleTextsOnChange;
-				_sampleTextsWatcher.Renamed += SampleTextsOnChange;
-				_sampleTextsWatcher.Deleted += SampleTextsOnChange;
-				_sampleTextsWatcher.EnableRaisingEvents = true;
+				if (_sampleTextsWatcher == null)
+				{
+					var path = Path.Combine(Path.GetDirectoryName(CurrentCollectionSettings.SettingsFilePath), "Sample Texts");
+					if (!Directory.Exists(path))
+						Directory.CreateDirectory(path);
+
+					_sampleTextsWatcher = new FileSystemWatcher { Path = path };
+					_sampleTextsWatcher.Created += SampleTextsOnChange;
+					_sampleTextsWatcher.Changed += SampleTextsOnChange;
+					_sampleTextsWatcher.Renamed += SampleTextsOnChange;
+					_sampleTextsWatcher.Deleted += SampleTextsOnChange;
+					_sampleTextsWatcher.EnableRaisingEvents = true;
+				}
 			}
 
-			var hasChanged = _sampleTextsChanged;
+			lock (_sampleTextsWatcher)
+			{
+				var hasChanged = _sampleTextsChanged;
 
-			// Reset the changed flag.
-			// NOTE: we are only resetting the flag if it was "true" when we checked in case the FileSystemWatcher detects a change
-			// after we check the flag but we reset it to false before we check again.
-			if (hasChanged) _sampleTextsChanged = false;
+				// Reset the changed flag.
+				// NOTE: we are only resetting the flag if it was "true" when we checked in case the FileSystemWatcher detects a change
+				// after we check the flag but we reset it to false before we check again.
+				if (hasChanged)
+					_sampleTextsChanged = false;
 
-			info.ContentType = "text/plain";
-			info.WriteCompleteOutput(hasChanged ? "yes" : "no");
+				info.ContentType = "text/plain";
+				info.WriteCompleteOutput(hasChanged ? "yes" : "no");
 
-			return true;
+				return true;
+			}
 		}
 
 		private void SampleTextsOnChange(object sender, FileSystemEventArgs fileSystemEventArgs)
```

Comment:
Should this method so some locking, since (although not public) it can get called outside any other lock by the FileSystemWatcher?